examples-gallery/plot_one_legged_time_trial.py

```diff
@@ -0,0 +1,351 @@
+                                                          knee_bot_act)
+ankle_top_pathway = me.LinearPathway(Co, P2)
+ankle_top_act = bm.FirstOrderActivationDeGroote2016.with_defaults('ankle_top')
+ankle_top_mus = bm.MusculotendonDeGroote2016.with_defaults('ankle_top',
+                                                           ankle_top_pathway,
+                                                           ankle_top_act)
+ankle_bot_pathway = me.LinearPathway(Co, P7)
+ankle_bot_act = bm.FirstOrderActivationDeGroote2016.with_defaults('ankle_bot')
+ankle_bot_mus = bm.MusculotendonDeGroote2016.with_defaults('ankle_bot',
+                                                           ankle_bot_pathway,
+                                                           ankle_bot_act)
+
+loads = (
+    knee_top_mus.to_loads() +
+    knee_bot_mus.to_loads() +
+    ankle_top_mus.to_loads() +
+    ankle_bot_mus.to_loads() +
+    [resistance, gravB, gravC, gravD]
+)
+
+kane = me.KanesMethod(
+    N,
+    (q1, q2),
+    (u1, u2),
+    kd_eqs=(
+        u1 - q1.diff(),
+        u2 - q2.diff(),
+        u3 - q3.diff(),
+        u4 - q4.diff(),
+    ),
+    q_dependent=(q3, q4),
+    configuration_constraints=holonomic,
+    velocity_constraints=holonomic.diff(me.dynamicsymbols._t),
+    u_dependent=(u3, u4),
+)
+bodies = (crank, foot, lower_leg, upper_leg)
+Fr, Frs = kane.kanes_equations(bodies, loads)
+
+muscle_diff_eq = sm.Matrix([
+    knee_top_mus.x[0, 0].diff() - knee_top_mus.rhs()[0, 0],
+    knee_bot_mus.x[0, 0].diff() - knee_bot_mus.rhs()[0, 0],
+    ankle_top_mus.x[0, 0].diff() - ankle_top_mus.rhs()[0, 0],
+    ankle_bot_mus.x[0, 0].diff() - ankle_bot_mus.rhs()[0, 0],
+])
+
+eom = kindiff.col_join(Fr + Frs).col_join(muscle_diff_eq).col_join(holonomic)
+
+state_vars = (
+    q1, q2, q3, q4, u1, u2, u3, u4,
+    knee_top_mus.x[0, 0],
+    knee_bot_mus.x[0, 0],
+    ankle_top_mus.x[0, 0],
+    ankle_bot_mus.x[0, 0],
+)
+
+num_nodes = 200
+h = sm.symbols('h', real=True, positive=True)
+
+# objective
+# minimize h
+# gradient of h wrt free variables: states, unknown_inputs, unknown_pars, h
+# so then there would be all zeros and a single 1 in the gradient for the last
+# entry
+
+
+def obj(free):
+    return free[-1]
+
+
+def gradient(free):
+    grad = np.zeros_like(free)
+    grad[-1] = 1.0
+    return grad
+
+
+# body segment inertia from https://nbviewer.org/github/pydy/pydy-tutorial-human-standing/blob/master/notebooks/n07_simulation.ipynb
+par_vals = {
+    Ar: 0.55,  # m^2, Tab 5.1, pg 188 Wilson 2004, Upright commuting bike
+    CD: 1.15,  # unitless, Tab 5.1, pg 188 Wilson 2004, Upright commuting bike
+    Cr: 0.006,  # unitless, Tab 5.1, pg 188 Wilson 2004, Upright commuting bike
+    G: 2.0,
+    IAzz: 0.0,
+    IBzz: 0.01,  # guess, TODO
+    ICzz: 0.101,  # lower_leg_inertia [kg*m^2]
+    IDzz: 0.282,  # upper_leg_inertia [kg*m^2],
+    J: 2*0.1524,  # from Browser Jason's thesis (rear wheel times 2)
```

Comment:
Could include the "2" above. Decide on definition of J (inertia of one or two wheels).